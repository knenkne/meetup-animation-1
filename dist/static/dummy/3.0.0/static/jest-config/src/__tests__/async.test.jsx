import React from 'react'
import axios from 'axios'
import { mount } from 'enzyme'
import MockAdapter from 'axios-mock-adapter'

import {
    AsyncComponentWithOneRequest,
    AsyncComponentWithTwoRequests
} from '../async-component'
import { multipleRequest } from '../../utils/async-mock'

const fetchData = (cb) => Promise.resolve('Success').then(cb) // асинхронный вызов в вашем компоненте

describe('Тест асинхронности', () => {

    let mock

    beforeEach(() => {
        mock = new MockAdapter(axios)
    })

    test('Получил в ответ "Success"', (done) => { // Внимание на принимаемый callback - done!
        function callback(data) {
            expect(data).toBe('Success')
            done()
        }

        fetchData(callback)
    })

    test('<AsyncComponentWithOneRequest />', async () => {
        const component = mount(
            <AsyncComponentWithOneRequest />
        )

        expect(component).toMatchSnapshot()

        const button = component.find('#getNewName')
        button.simulate('click')

        const responseShow = [
            ['GET', '/getFirstName', {}, 200, { newName: 'John' }]
        ]

        await multipleRequest(mock, responseShow)

        component.update()
        expect(component).toMatchSnapshot()
    })

    test('<AsyncComponentWithTwoRequests />', async () => {

        const component = mount(
            <AsyncComponentWithTwoRequests />
        )

        expect(component).toMatchSnapshot()

        const button = component.find('#getNewPerson')
        button.simulate('click')

        const responseShow = [
            ['GET', '/getFirstName', {}, 200, { firstName: 'John' }],
            ['GET', '/getSecondName', {}, 200, { secondName: 'Konstantin' }]
        ]

        await multipleRequest(mock, responseShow)

        component.update()
        expect(component).toMatchSnapshot()
    })

})
