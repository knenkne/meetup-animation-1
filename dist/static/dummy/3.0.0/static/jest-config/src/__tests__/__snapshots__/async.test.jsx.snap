// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест асинхронности <AsyncComponentWithOneRequest /> 1`] = `
<AsyncComponentWithOneRequest>
  <div>
    <button
      id="getNewName"
      onClick={[Function]}
    >
      Get New Name
    </button>
    <h1>
      Nikola
    </h1>
  </div>
</AsyncComponentWithOneRequest>
`;

exports[`Тест асинхронности <AsyncComponentWithOneRequest /> 2`] = `
<AsyncComponentWithOneRequest>
  <div>
    <button
      id="getNewName"
      onClick={[Function]}
    >
      Get New Name
    </button>
    <h1>
      John
    </h1>
  </div>
</AsyncComponentWithOneRequest>
`;

exports[`Тест асинхронности <AsyncComponentWithTwoRequests /> 1`] = `
<AsyncComponentWithTwoRequests>
  <div>
    <button
      id="getNewPerson"
      onClick={[Function]}
    >
      Get New Person
    </button>
    <h1>
      Nikola Tesla
    </h1>
  </div>
</AsyncComponentWithTwoRequests>
`;

exports[`Тест асинхронности <AsyncComponentWithTwoRequests /> 2`] = `
<AsyncComponentWithTwoRequests>
  <div>
    <button
      id="getNewPerson"
      onClick={[Function]}
    >
      Get New Person
    </button>
    <h1>
      John Konstantin
    </h1>
  </div>
</AsyncComponentWithTwoRequests>
`;
